torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,155 @@
+    """
+    Transformations for adding noise to images.
+    """
+
+    def __init__(self, noise_variance=(0.001, 0.05), **kwargs):
+        """
+
+        :param noise_variance: The variance of the noise added to the  image.
+        :param kwargs: Arguments of the super class.
+        """
+        self.device = kwargs["device"]
+        self.noise_variance = noise_variance
+        DictTransform.__init__(self, **kwargs)
+
+    def transform(self, data):
+        std = torch.rand(data.size(0) if data.ndim > 5 else 1, device=data.device, dtype=data.dtype)
+        return data + torch.randn_like(data) * std
+
+
+class BlurDictTransform(DictTransform):
+    """Transformation for adding Blur to images."""
+
+    def __init__(self, channels=1, kernel_size=(3, 3, 3), sigma=1, **kwargs):
+        """
+        :param channels: Amount of channels of the input data.
+        :param kernel_size: Size of the convolution kernel.
+        :param sigma: Standard deviation.
+        """
+        self.channels = channels
+        self.sigma = [sigma, sigma, sigma]
+        self.kernel_size = kernel_size
+        self.device = kwargs["device"]
+        DictTransform.__init__(self, **kwargs)
+        # code from: https://discuss.pytorch.org/t/is-there-anyway-to-do-gaussian-filtering-for-an-image-2d-3d-in-pytorch/12351/10
+        # initialize conv layer.
+        kernel = 1
+        # todo add dynamic padding for higher kerner_size
+        #
+        self.meshgrids = torch.meshgrid(
+            [
+                torch.arange(size, dtype=torch.float32)
+                for size in self.kernel_size
+            ]
+        )
+        for size, std, mgrid in zip(self.kernel_size, self.sigma, self.meshgrids):
+            mean = (size - 1) / 2
+            kernel *= 1 / (std * math.sqrt(2 * math.pi)) * torch.exp(-((mgrid - mean) / std) ** 2 / 2)
+
+        # Make sure sum of values in gaussian kernel equals 1.
+        kernel = kernel / torch.sum(kernel)
+
+        # Reshape to depthwise convolutional weight
+        kernel = kernel.view(1, 1, *kernel.size())
+        kernel = kernel.repeat(self.channels, *[1] * (kernel.dim() - 1))
+
+        kernel = kernel.to(self.device)
+        self.weight = kernel
+        self.conv = F.conv3d
+
+    def transform(self, data):
+        vol = make_5d(data)
+        vol = self.conv(vol, weight=self.weight, groups=self.channels, padding=1)
+        return vol
+
+
+class CroppingTransform(DictTransform):
+    """
+    Transformation for the cropping of 4D or 5D Volumes.
+    """
+
+    def __init__(self, size=20, position=0, **kwargs):
+        """
+        :param size: Size of the crop.
+        :param position: Middle point of the cropped region.
+        :param kwargs: Arguments for super class.
+        """
+        DictTransform.__init__(self, **kwargs)
+        self.size = size
+        self.position = position
+
+    def transform(self, data):
+        data = self.get_center_crop(data, self.size)
+        return data
```

Comment:
```suggestion
        return self.get_center_crop(data, self.size)
```

Removing temporary variables can actually give significant speedup I think.